packages/core/src/telemetry/detached-flush.test.ts

```diff
@@ -0,0 +1,37 @@
+import child_process from "node:child_process";
+import { rmSync, writeFileSync } from "node:fs";
+import { tmpdir } from "node:os";
+import path from "node:path";
+import process from "node:process";
+import { expect, test } from "vitest";
+
+test("detached flush script should run without error", async ({ common }) => {
+  const events = Array.from({ length: 5 }, () => ({
+    event: "test",
+    payload: {},
+  }));
+
+  const telemetryEventsFilePath = path.join(tmpdir(), "events.json");
```

Comment:
Can just use the handy `tmpdir()` function here to avoid potentially leaving undesired files in the repo.